api/document/tree.py

```diff
@@ -125,45 +217,51 @@ def nested_set_renumber(self, left=1, bulk_create=True):
         if bulk_create:
             self._bulk_create()
 
-    def _bulk_create(self):
+    def _bulk_create(self) -> None:
         DocNode.objects.bulk_create(node.model for node in self.walk())
 
-    def next_sort_order(self):
+    def next_sort_order(self) -> int:
         return self.tree.out_degree(self.identifier)
```

Comment:
I _think_ this returns an `int`. According to the [`out_degree` documentation](https://networkx.github.io/documentation/latest/reference/classes/generated/networkx.DiGraph.out_degree.html#networkx.DiGraph.out_degree), it could also return an iterator if multiple nodes are requested, but it seems like we're only requesting one node here.

Reply:
You got it! We use this number to tag the edges between nodes. This allows us to fetch child nodes in order (as trees don't have an inherent order but documents do).